tests/pkg_edhoc_c/initiator.c

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright (C) 2021 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       EDHOC coap initiator implementation
+ *
+ * @author      Timothy Claeys <timothy.claeys@inria.fr>
+ * @author      Francisco Molina <francois-xavier.molina@inria.fr>
+ */
+
+#include <stdio.h>
+
+#include "net/ipv6.h"
+#include "net/nanocoap_sock.h"
+#include "shell.h"
+
+#include "edhoc/edhoc.h"
+#include "edhoc_keys.h"
+#include "wolfssl/wolfcrypt/sha256.h"
+
+#define ENABLE_DEBUG        0
+#include "debug.h"
+
+#define COAP_BUF_SIZE     (512U)
+
+extern void print_bstr(const uint8_t *bstr, size_t bstr_len);
+extern int edhoc_setup(edhoc_ctx_t *ctx, edhoc_conf_t *conf, edhoc_role_t role,
+                       cose_key_t *auth_key, cred_id_t *cred_id, rpk_t *rpk,
+                       void *hash_ctx);
+extern void edhoc_oscore_exporter(edhoc_ctx_t *ctx, uint8_t *secret, size_t secret_len,
+                                  uint8_t *salt, size_t salt_len);
+
+static edhoc_ctx_t _ctx;
+static edhoc_conf_t _conf;
+static rpk_t _rpk;
+static cred_id_t _cred_id;
+static cose_key_t _auth_key;
+static wc_Sha256 _wc_sha256;
+static uint8_t _method;
+static uint8_t _suite;
+
+static ssize_t _send(coap_pkt_t *pkt, size_t len, char *addr_str, uint16_t port)
```

Comment:
@kaspar030 maybe you can take a look at this section?

Reply:
I seems ok. Is there a reason why you're not using `sock_udp_str2ep()`?

... Looking into that, I see that your function does actually verify whether the netif is a valid gnrc interface, so this cool. We should make this functionality available to sock at some point.